# Using _encode__sequences.py_

## Contents 
#### (the functions contained within _encode__sequences.py_)

1. encode_sequence_with_method()
1. one_hot_encode_sequence()
1. resize_array()

## How to use them

### 2. one_hot_encode_sequence()

#### Import the function

In [15]:
from encode_sequences import one_hot_encode_sequence as one_hot_encode

#### Define the nucleotide sequence to be encoded

In [22]:
sequence = 'TGCATTTTTTTCACATCTATTGGCTACTAATCAAAGGGACTCGGTGGATTTAATTCTGTTGATTCCGAAGCCTCTTATGTGCTCAAGTTTGGGTAGAGGTTACGGCTGTT'
print('Length of sequence: %s' %(len(sequence)))


Length of sequence: 110


In [17]:
one_hot_sequence = one_hot_encode(sequence)
one_hot_sequence

[[0, 1, 0, 0],
 [0, 0, 1, 0],
 [0, 0, 0, 1],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [1, 0, 0, 0],
 [0, 0, 0, 1],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [0, 1, 0, 0],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 1, 0],
 [0, 0, 1, 0],
 [0, 0, 0, 1],
 [0, 1, 0, 0],
 [1, 0, 0, 0],
 [0, 0, 0, 1],
 [0, 1, 0, 0],
 [1, 0, 0, 0],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [1, 0, 0, 0],
 [1, 0, 0, 0],
 [1, 0, 0, 0],
 [0, 0, 1, 0],
 [0, 0, 1, 0],
 [0, 0, 1, 0],
 [1, 0, 0, 0],
 [0, 0, 0, 1],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [0, 0, 1, 0],
 [0, 0, 1, 0],
 [0, 1, 0, 0],
 [0, 0, 1, 0],
 [0, 0, 1, 0],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [1, 0, 0, 0],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [0, 1, 0, 0],
 [0, 0, 1, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 1, 0],
 [1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [0, 0, 0, 1],
 [0, 0, 1,

In [23]:
print('Length of encoded sequence: %s' %(len(one_hot_sequence)))

Length of sequence: 110


* Note that the length of the nucleotide sequence is maintained upon encoding.
* Now suppose I want to resize my encoded sequence (this will be important later as all of the 2D array inputs into the neural network should be of the same dimensions). This brings us on to the next function ...

### 3. resize_array()

#### Import the function

In [25]:
from encode_sequences import resize_one_hot_seq as resize

* Suppose I want to define my desired maximum length of the encoded nucleotide sequence to be 100 base pairs. 
* The sequence ``oligo`` I have defined in part 2 is of length 110, so 10 base vectors will be trimmed off of the front of the one-hot encoded 2D array.
* If I wanted to trim the base vectors off the end (right hand side) I would define ``edit_front = False``

In [24]:
resized_one_hot_seq = resize(one_hot_sequence, max_length=100, edit_front=True)

print('Length of resized encoded sequence: %s' %(len(resized_one_hot_seq)))

Length of resized encoded sequence: 100
